be1-go/message2/query/query.go

```diff
@@ -0,0 +1,67 @@
+package query
+
+import (
+	"encoding/json"
+	"student20_pop/message2/query/method"
+
+	"golang.org/x/xerrors"
+)
+
+// Query ...
+type Query struct {
+	Method string
```

Comment:
I have a suggestion based also on my comment for the `JsonRpcMessage` struct/interface.

Can we add the `Id` here and make it optional (or its semantic equivalent)? The JSON-Schema will validate anyways that it's present where it's needed and absent where it's not. Only the marshaling logic will have to be aware that the `Id` should be absent in some cases.

If so, the *only* difference between the methods will be the params. Thus, each method object could be defined as implementing a `QueryParams` interface which is implemented like the `JsonRpcMessage` suggestion.
This would also be trivial to use, because based on the `Method` field, we can simply cast the `QueryParams` to the right type.

The beauty of this solution is that it also massively simplifies the current method objects, as they can all be aliases of just 2 types (params = a channel, and params = channel+message), which minimizes the amount of repetition.

Reply:
I'd rather not. IMO it's only adding useless complexity, as the actual implementation already simple and explicit.